apps/condo/domains/property/components/panels/Builder/MapConstructor.ts

```diff
@@ -0,0 +1,490 @@
+                floors.unshift(floorIndex)
+            }
+        }
+        return floors
+    }
+
+    get isEmpty (): boolean {
+        return this.map.sections.length === 0
+    }
+
+    get sections (): BuildingSection[] {
+        return this.map.sections
+    }
+    
+    protected getUnitIndex (unitId: string): IndexLocation {
+        const result = { section: -1, floor: -1, unit: -1 }
+        try {
+            this.map.sections.forEach((section, sectionIdx) => {
+                section.floors.forEach((floor, floorIdx) => {
+                    const unitIdx = floor.units.findIndex(unit => unit.id === unitId)
+                    if (unitIdx !== -1) {
+                        result.section = sectionIdx
+                        result.floor = floorIdx
+                        result.unit = unitIdx
+                        throw new Error('Found')
+                    }
+                })
+            })
+        } catch (_) {
+            //
+        }
+        return result
+    }
+
+    public getUnitInfo (id: string): BuildingUnit {
+        const newUnit = { id: null, type: 'unit', label: '', floor: '', section: '' }
+        if (!id) {
+            return newUnit
+        }
+        const unitIndex = this.getUnitIndex(id)
+        if (unitIndex.unit === -1) {
+            return newUnit
+        }
+        const { label, type } = this.map.sections[unitIndex.section].floors[unitIndex.floor].units[unitIndex.unit]
+        return {
+            id,
+            label, 
+            type,
+            section: this.map.sections[unitIndex.section].id,
+            floor: this.map.sections[unitIndex.section].floors[unitIndex.floor].id,
+        }
+    }
+
+    public  getSectionOptions (): BuildingSelectOption[] {
+        const options = this.sections.map(section => ({ id: section.id, label: section.name }))
+        return options
+    }
+    
+    public getSectionFloorOptions (id: string): BuildingSelectOption[] {
+        const foundSection = this.map.sections.find(section => section.id === id)
+        if (!foundSection) {
+            return []
+        }
+        const options = foundSection.floors.map(floor => ({ id: floor.id, label: floor.name }))
+        return options
+    }
+    
+}
+
+
+class MapEdit extends MapView {
+
+    constructor (map: Maybe<BuildingMap>, updateMap: Maybe<(map: BuildingMap) => void>) {
+        super(map)
+        this.updateMap = updateMap
+    }   
+
+    get nextUnitNumber (): number {
+        const result = Math.max(0, ...this.map.sections
+            .map(section => section.floors
+                .map(floor => floor.units
+                    .map(unit => !isNaN(+unit.label) ? Number(unit.label) : 0)))
+            .flat(2)) + 1
+        return result
+    }
+
+    private updateUnitNumbers (unitFrom: BuildingUnit): void {
+        const { id, label } = unitFrom
+        if (isNaN(+label)) {
+            return
+        }
+        let started = false
```

Comment:
??

Reply:
Looks cool :) For now - replaced
```js
+label
```
to 
```js 
Number(label)
```
Need to change some logic after finding out that Unit can have name